charms/jupyter-ui/tests/unit/test_operator.py

```diff
@@ -28,6 +29,24 @@ def harness() -> Harness:
 class TestCharm:
     """Test class for JupyterUI."""
 
+    def test_spawner_ui(self):
+        """Test spawner UI.
+
+        spawner_ui_config.yaml contains a number of changes that were done for Charmed
+        Kubeflow. This test is to validate those. If it fails, spawner_ui_config.yaml
+        should be reviewed and changes to this tests should be made, if required.
+        """
+        spawner_ui_config = yaml.safe_load(Path("./src/spawner_ui_config.yaml").read_text())
+
+        # test for default configurations
+        # only single configuration value is currently set in the list of values
+        config_value = spawner_ui_config["spawnerFormDefaults"]["configurations"]["value"]
+        assert config_value == ["access-ml-pipeline"]
+
+        # test for images added in addition to upstream
+        image_list = spawner_ui_config["spawnerFormDefaults"]["image"]["options"]
+        assert "swr.cn-south-1.myhuaweicloud.com/mindspore/jupyter-mindspore:v1.6.1" in image_list
```

Comment:
imo this test is too specific.  In general, we want to protect against accidentally removing the mindspore image during upgrades, so we need to assert that we have **a** mindspore image not **this specific tag** of the mindspore image.  We effectively are protecting against a human error of removing the image.

The check in the current PR asserts we have **a specific tagged image**, which means whenever we upgrade our kubeflow version we will also need to change this test.  From a maintenance standpoint that is a chore and opens us up to other mistakes here.  In general, I don't think we want the standard behaviour for every upgrade to be to edit the test that makes sure our upgrade happened right.  

imo, we can get the same protection with a lower maintenance test as suggested below.  In this test, we assert that we have **a version of** the mindspore image, not a specific tag of one.  This way we are still protected against accidentally removing the image, but nobody needs to edit this test every time.

```suggestion
        image_list = spawner_ui_config["spawnerFormDefaults"]["image"]["options"]
        assert any("swr.cn-south-1.myhuaweicloud.com/mindspore/jupyter-mindspore" in image for image in image_list)
```

Reply:
I will record a disagreement with not checking version of Mindspore image.